pypsa/networkclustering.py

```diff
@@ -351,14 +351,19 @@ def get_buses_linemap_and_lines(
     lines, linemap_p, linemap_n, linemap, lines_t = aggregatelines(
         network, buses, interlines, line_length_factor, with_time
     )
+    # network can be reduced to a set of isolated nodes in course of clustering (e.g. Rwanda)
+    if lines.empty:
+        lines_res = lines.drop(columns=["bus0_s", "bus1_s"]).reset_index(drop=True)
```

Comment:
Why are there different options for `reset_index()` between the two cases?

Reply:
The reason of troubles was a conflict between columns names and index names. The `lines` dataframe if being empty obtains an additional index after `aggregatelines()` applied, and this index contains `bus0_s` and `bus1_s`. Further, when trying to reset the index, there is a conflict between the columns and index names in case pandes default `drop=False`.

To avoid this names conflict, it might be enough to simply drop `bus0_s` and `bus1_s` columns, but I aimed to also keep structure of `lines` dataframe same as in the basic workflow version